### DISCUSSION TITLE
afterScenario does not work in Async mode

### DISCUSSION THREAD

**Discussion by suchitrak** [webdriverio/webdriverio#8613] [Q&A]
I am slowing migrating webdriverio tests to Async. My package.json file still has @wdio/sync library. Using V7 version of wdio.  The custom commands that have been moved to Async works as expected if invoked from tests or when the hook methods are not made Async. 
But If I change the hooks to Async, the custom commands are not invoked synchronously even after adding the right awaits

Question: Do the hooks have to be Asynchronous after the @wdio/sync is deleted from the repository?

config File
```
async afterScenario(world, result, context) {
        await console.log("Start Clean UP")
            cleanUp(browser);
            await console.log("End Clean UP")
    }
```

Clean Up utility
```
async cleanUp(browser) {

               await browser.config.idsOfItemsCreated.files.forEach(async file => {
                await console.log(`Deleting File as part of clean up ${file.id}`);
                await browser.config.currentUserName = file.currentUserName;
                await browser.box_deleteItem(file.id, 'file');
               await console.log(`Completed Deleting File as part of clean up ${file.id}`);
            });
}
```

Custom command

```
module.exports = browser => {
browser.addCommand('box_deleteItem', async (itemID, itemType) => {
 
       // Invoke API using fetch library to delete an item. This works as expected if invoked from the tests instead of invoking from hook

await console.log("Deleted item")
    });


}

```

Prints:
Start Clean UP
Deleting File as part of clean up 
 browser.box_deleteItem(file.id, 'file');
End Clean UP

Lastly prints the results but clean up is not done.

**Comment by christian-bromann** [comment]
> Question: Do the hooks have to be Asynchronous after the @wdio/sync is deleted from the repository?

Yes.

**Comment by suchitrak** [comment]
So should it be something like this  here - https://webdriver.io/docs/configurationfile/

```
afterScenario: async function (world, result, context) {
     await console.log("afterScenario start")
     await browser.customCommandMethod()
},

```

**Reply by christian-bromann** [reply]
yes
